src/main/test/com/epam/esm/service/GiftCertificateServiceTest.java

```diff
@@ -0,0 +1,109 @@
+package com.epam.esm.service;
+
+import com.epam.esm.dao.GiftCertificateDao;
+import com.epam.esm.dao.GiftCertificateTagDao;
+import com.epam.esm.dao.TagDao;
+import com.epam.esm.dto.GiftCertificateDto;
+import com.epam.esm.entity.GiftCertificate;
+import com.epam.esm.exception.BadEntityException;
+import com.epam.esm.exception.EntityNotExistsException;
+import com.epam.esm.utils.GiftCertificateDtoMapper;
+import com.epam.esm.validator.GiftCertificateValidator;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.mockito.Mockito;
+import org.springframework.core.env.Environment;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyLong;
+import static org.mockito.Mockito.*;
+
+public class GiftCertificateServiceTest {
```

Comment:
We need more tests for the service layer. Per requirements test coverage for a service layer should be no less than 80 percent. Right now it's only 20%:
![image](https://user-images.githubusercontent.com/17936204/131496243-f8cc59c9-b660-4bb1-8cef-7f3cf00fe2c7.png)

Reply:
Now it has 91% test coverage according to IntelliJ IDEA tools